packages/ecs/src/QueryFunctions.test.tsx

```diff
@@ -0,0 +1,181 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { act, render } from '@testing-library/react'
+import assert from 'assert'
+import React, { useEffect } from 'react'
+import { ComponentMap, defineComponent, setComponent } from './ComponentFunctions'
+import { destroyEngine, startEngine } from './Engine'
+import { Entity, UndefinedEntity } from './Entity'
+import { createEntity, removeEntity } from './EntityFunctions'
+import { Query, defineQuery, useQuery } from './QueryFunctions'
+
+function assertArrayEqual<T>(A: Array<T>, B: Array<T>, err = 'Arrays are not equal') {
```

Comment:
are these helpers necessary? does `assert.deepEqual` not work for just two arrays?

Reply:
I don't know, honestly. I will try it :writing_hand:

Reply:
Well, apparently I did try it before.
It doesn't work with `assert.deepEqual` directly on the arrays without iterating :shrug: